sql/core/src/main/scala/org/apache/spark/sql/execution/local/ConvertToSafeNode.scala

```diff
@@ -27,6 +27,10 @@ case class ConvertToSafeNode(conf: SQLConf, child: LocalNode) extends UnaryLocal
 
   private[this] var convertToSafe: Projection = _
 
+  override def prepare(): Unit = {
+    child.prepare()
```

Comment:
Actually then we could just implement it in LocalNode itself since it can traverse the children.